docs/plugin/interfaces/guidelines.md

```diff
@@ -0,0 +1,383 @@
+Interface Definition
+====================
+
```

Comment:
Adding a few generic comment.
First I really like what is already here.

- I think we might be missing some guidelines for how we recommend Interfaces are tagged (e.g @text:keep, @alt, ..) to properly support JsonRPC. 
- Also, we might aligning on using or camel case vs Pascal case for methods, at the same time having it consistent across JsonRPC and Com-RPC
- A section with respect to versioning of interfaces may make sense. When an interface is extended, how is the version updated.